src/sql/workbench/browser/modelComponents/inputbox.component.ts

```diff
@@ -142,6 +143,14 @@ export default class InputBoxComponent extends ComponentBase implements ICompone
 	public validate(): Thenable<boolean> {
 		return super.validate().then(valid => {
 			this.inputElement.validate();
+
+			// set aria label based on validity of input
+			if (valid) {
+				this.inputElement.setAriaLabel(this._defaultAriaLabel);
+			} else {
+				this.inputElement.setAriaLabel((this._defaultAriaLabel ? this._defaultAriaLabel + '. ' : '') + this.inputElement.inputElement.validationMessage);
```

Comment:
Nesting a ternary in an if-statement seems questionable.  Maybe worth unpacking the logic a bit.

Reply:
@Benjin how come?

Reply:
Better readability -> easier maintainability.  Mostly because I think branching logic off of a `string?`'s state then immediately using it in a composite is funky.  More lines of code sometimes means easier to trace.